File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/context/XmlFileContext_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "XmlFileContext_ViewModelTestEnvironment.h"
#include <string>
#include "XmlFileContext_ViewModel.h"
#include "XmlFileContext_ViewModelTestEnvironmentImpl.h"

namespace context
{
  class XmlFileContext_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<XmlFileContext_ViewModel> sut;
    std::shared_ptr<XmlFileContext_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    // Given Helper Definitions
    std::string XmlFile_NoExternalFile = R"(<MyXML>
  <Inner></Inner>
</MyXML>)";
    virtual void given_XmlFile_NoExternalFile();
    virtual void given_XmlFile_ExternalFile();
  protected:
    void SetUp() override;
  };
  void XmlFileContext_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<XmlFileContext_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  /// Scenario: XmlFile_NoExternalFile
  ///   given: XmlFile_NoExternalFile
  ///    when:
  ///    then:
  TEST_F(XmlFileContext_ViewTests, XmlFile_NoExternalFile_given_XmlFile_NoExternalFile_when_then_)
  {
    this->given_XmlFile_NoExternalFile();
    this->BuildSut();
  }
  /// Scenario: XmlFile_ExternalFile
  ///   given: XmlFile_ExternalFile
  ///    when:
  ///    then:
  TEST_F(XmlFileContext_ViewTests, XmlFile_ExternalFile_given_XmlFile_ExternalFile_when_then_)
  {
    this->given_XmlFile_ExternalFile();
    this->BuildSut();
  }
  void XmlFileContext_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void XmlFileContext_ViewTests::given_XmlFile_NoExternalFile()
  {
    this->testEnvironment->SetXmlFileContext(this->XmlFile_NoExternalFile);
  }
  void XmlFileContext_ViewTests::given_XmlFile_ExternalFile()
  {
    this->testEnvironment->SetXmlExternalFileContext(std::string("XmlFile_ExternalFile.xml"));
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/context/XmlFileContext_ViewTests.cs
/// <filename>
///     XmlFileContext_ViewTests.cs
/// </filename>
using context;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class XmlFileContext_ViewTests
{
    private context.XmlFileContext_ViewModel sut;

    private context.XmlFileContext_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new context.XmlFileContext_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    /*

    Scenario: XmlFile_NoExternalFile
      given: XmlFile_NoExternalFile
       when:
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void XmlFile_NoExternalFile_given_XmlFile_NoExternalFile_when_then_()
    {
        this.given_XmlFile_NoExternalFile();
        this.BuildSut();
    }

    /*

    Scenario: XmlFile_ExternalFile
      given: XmlFile_ExternalFile
       when:
       then:
   */
    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void XmlFile_ExternalFile_given_XmlFile_ExternalFile_when_then_()
    {
        this.given_XmlFile_ExternalFile();
        this.BuildSut();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    // Given Helper Definitions
    private string XmlFile_NoExternalFile = @"<MyXML>
  <Inner></Inner>
</MyXML>";

    public virtual void given_XmlFile_NoExternalFile()
    {
        this.testEnvironment.SetXmlFileContext(this.XmlFile_NoExternalFile);
    }

    public virtual void given_XmlFile_ExternalFile()
    {
        this.testEnvironment.SetXmlExternalFileContext("XmlFile_ExternalFile.xml");
    }
}
